src/shogun/regression/svr/LibLinearRegression.cpp

```diff
@@ -199,7 +209,7 @@ void CLibLinearRegression::solve_l2r_l1l2_svr(SGVector<float64_t>& w, const libl
 		prob->x->add_to_dense_vec(beta[i], i, w.vector, w_size);
 
 		if (prob->use_bias)
-			w.vector[w_size]+=beta[i];
+            w.vector[w_size+1]+=beta[i];
```

Comment:
why `w_size+1`? if w has `w_size+1` elements, the last element is `w.vector[w_size]`

Reply:
The original issue described in #4382 is due to a mismatch of `w_size` and `prob.x.num_features`, the former included bias, whereas the latter did not. This results in an assertion error when performing vectorised operations with `prob.x`, i.e. `prob->x->add_to_dense_vec(...)`.

 As a fix I set `w_size` to the size of weights, excluding the bias term. 
```
if (prob->use_bias)
    w_size = prob->n - 1;
else w_size = prob->n;
```
Therefore, the bias is in the `w_size+1` position. It seemed to be the cleanest fix! I could also have written something like this:
```
if (prob->use_bias)
{
    prob->x->add_to_dense_vec(beta[i], i, w.vector, prob->n-1);
    w.vector[prob->n]+=beta[i];
}
else prob->x->add_to_dense_vec(beta[i], i, w.vector, prob->n);
```

Reply:
Thanks for clarifying. The two code snippets indeed are equivalent, but I am still confused.
When you set `w_size` to the size of weights (without bias), `w.vector[0] ... w.vector[w_size-1]` are weights, and `w.vector[w_size]` is bias. Maybe you have different idea?

Reply:
Yes, that is right! I seem to have gotten confused, thanks for pointing that out. The bias worked before, but could lead to undefined behaviour.. I have fixed it now!